### PR TITLE
added PictureCard component to the library

### PR DESCRIPTION
**Pull request by sahandfarahani**
PictureCardwithimage
PictureCardwithtags

### CODE REVIEW

File: src/Containers/PictureCard.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { MainInterface, ResponsiveInterface } from '@Utils/BaseStyles';
+import styled from 'styled-components';
+import { StyledIcon } from 'styled-icons/types';
+import { flex, position } from '@Utils/Mixins';
+import { Card as C } from './Card';
+
+const Card = styled(C)`
+    ${'max-width:300px;'}
```
Comment (ralph-dev) [inline src/Containers/PictureCard.tsx:9]: Why not do,
max-width: 300px;

File: src/Containers/PictureCard.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import { flex, position } from '@Utils/Mixins';
+import { Card as C } from './Card';
+
+const Card = styled(C)`
+    ${'max-width:300px;'}
+`;
+
+interface PictureCardProps extends MainInterface, ResponsiveInterface {
+    image: string;
+    tags: { icon: StyledIcon; text: string }[];
```
Comment (ralph-dev) [inline src/Containers/PictureCard.tsx:14]: Abstract icon and text object into an interface so you can write

TagProps[]

File: src/Containers/PictureCard.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                                <Tag>
+                                    {icon && <Icon as={icon} hasText={text} />}
+                                    {text}
+                                </Tag>
+                            ),
+                        )}
+                    </Tags>
+                ) : null}
+            </ImageWrapper>
+            {!children ? null : <Content>{children}</Content>}
```
Comment (ralph-dev) [inline src/Containers/PictureCard.tsx:40]: Just do

Children && content

File: src/Containers/PictureCard.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    children,
+    image,
+    tags,
+    ...cardProps
+}): React.ReactElement => {
+    return (
+        <Card {...cardProps}>
+            <ImageWrapper>
+                <Image src={image} />
+                {tags.length ? (
```
Comment (nikitasheremet) [inline src/Containers/PictureCard.tsx:27]: You can just do tags && (<Tags>....</Tags>)

**Comment by sahandfarahani** [comment]
did all the changes

File: src/Containers/PictureCard.tsx (hunk 4 above)
Comment (sahandfarahani) [reply src/Containers/PictureCard.tsx]: done

File: src/Containers/PictureCard.tsx (hunk 3 above)
Comment (sahandfarahani) [reply src/Containers/PictureCard.tsx]: all done

File: src/Containers/PictureCard.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+const Card = styled(C)`
+    max-width: 300px;
+`;
+
+interface TagProps {
+    icon: StyledIcon;
+    text: string;
+}
+
+interface PictureCardProps extends MainInterface, ResponsiveInterface {
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:17]: Please export this

File: src/Containers/PictureCard.tsx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    image: string;
+    tags: TagProps[];
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:19]: Tags should be optional

File: src/Containers/PictureCard.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { MainInterface, ResponsiveInterface } from '@Utils/BaseStyles';
+import styled from 'styled-components';
+import { StyledIcon } from 'styled-icons/types';
+import { flex, position } from '@Utils/Mixins';
+import { Card as C } from './Card';
+
+const Card = styled(C)`
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:8]: This is evil, remove it or make max-width a prop

File: src/Containers/PictureCard.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+export const PictureCard: React.FC<PictureCardProps> = ({
+    children,
+    image,
+    tags,
+    ...cardProps
+}): React.ReactElement => {
+    return (
+        <Card {...cardProps}>
+            <ImageWrapper>
+                <Image src={image} />
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:31]: Let's add some optional props for this like width, height, alt

File: src/Containers/PictureCard.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        'absolute',
+        '0 0 10px 10px',
+        'auto',
+        'auto',
+        0,
+        0,
+    )}
+    font-size: 0.75rem;
+    font-weight: bold;
+    color: #3c3c3c;
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:72]: Toss this in theme pls

File: src/Containers/PictureCard.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+`;
+
+const Tag = styled.span`
+    ${flex('row', 'center')}
+    display: inline-flex;
+    background-color: white;
+    box-shadow: 0 0 3px rgba(0, 0, 0, 0.3);
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:79]: Try using depth in theme (if it looks ass, then leave it I guess). Also make a prop called flat to make this conditional

**Comment by fpunny** [comment]
Side question, but where is the spacing between the image and card coming from?

**Comment by sahandfarahani** [comment]
> Side question, but where is the spacing between the image and card coming from?

its the padding inherited from Card component I'm going to put it to "0" to see how it will look

**Comment by ralph-dev** [comment]
@sahandfarahani Make sure to post an updated screenshot iwth the new padding and let's get this merged before the EOD

**Comment by sahandfarahani** [comment]
PictureCardwithimage
PictureCardwithtags

**Comment by sahandfarahani** [comment]
photo_2020-04-17 16 32 31
photo_2020-04-17 16 32 36

File: stories/Containers/PictureCard.stories.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { PictureCard, Paragraph, SmallText } from '../../src';
+import { Dog, Poo } from 'styled-icons/fa-solid';
```
Comment (fpunny) [inline stories/Containers/PictureCard.stories.js:4]: Can you import these individually? Importanting from the root increases the build time by 10+ seconds (I wish we used react-icons instead)

File: src/Containers/PictureCard.tsx [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+    );
+};
+
+interface CardProps{
+    width?:string;
+    height?:string;
+}
+
+const Card = styled(C)<CardProps>`
+    width: ${({width}) => width ? width : '300px'};
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:60]: no default width, have it auto or something

File: src/Containers/PictureCard.tsx [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+    height:${({height}) => height ? height : 'auto'};
+    padding:4px;
```
Comment (fpunny) [inline src/Containers/PictureCard.tsx:62]: No padding and border-radius should come from theme

File: stories/Containers/PictureCard.stories.js (hunk 11 above)
Comment (sahandfarahani) [reply stories/Containers/PictureCard.stories.js:4]: done

File: src/Containers/PictureCard.tsx (hunk 12 above)
Comment (sahandfarahani) [reply src/Containers/PictureCard.tsx:60]: done

File: src/Containers/PictureCard.tsx (hunk 13 above)
Comment (sahandfarahani) [reply src/Containers/PictureCard.tsx:62]: It was coming theme padding from Card.tsx If I don't change it, the spacing between Card and Picture will come back

**Pull request merged**